src/usePreviousDistinct/usePreviousDistinct.ts

```diff
@@ -0,0 +1,33 @@
+import { useRef, useState } from 'react';
+import { useUpdateEffect } from '..';
+import { isStrictEqual } from '../util/const';
+
+export type TPredicate = <T>(prev: T, next: any) => next is typeof prev;
```

Comment:
Why `T` prefixed?

Reply:
I saw that interfaces have `I` prefix. I presumed I was just following that pattern, but I can revert if not!

Reply:
I've forgot to change this rule - i prefer to avoid prefixing for soime time now.
All new code lack of prefixes.